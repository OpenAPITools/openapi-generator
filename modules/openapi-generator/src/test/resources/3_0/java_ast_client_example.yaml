openapi: 3.0.0
info:
  version: 01.01.00
  title: APITest API documentation.
  termsOfService: http://api.apitest.com/party/tos/
servers:
  - url: https://api.apitest.com/v1
paths:
  /polygon:
    get:
      summary: Get the polygon!
      operationId: ping
      responses:
        '201':
          description: Created
          content:
            application/json:
              schema:
                type: string
                description: Polygon
        '400':
          description: Bad Request
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Polygon'
      parameters: []
components:
  schemas:
    Polygon:
      title: Polygon
      description: A Polygon!
      type: object
      properties:
        name:
          type: string
        sides:
          type: array
          items:
            $ref: '#/components/schemas/Side'
    Side:
      type: object
      properties:
        length:
          $ref: '#/components/schemas/ValueWithUnits'
    ValueWithUnits:
      type: object
      properties:
        unit:
          type: string
        magnitude:
          type: number




