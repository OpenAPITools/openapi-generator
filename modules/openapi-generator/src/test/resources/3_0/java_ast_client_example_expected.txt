public class Polygon {

    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    private List<Side> sides;

    public List<Side> getSides() {
        return sides;
    }

    public void setSides(List<Side> sides) {
        this.sides = sides;
    }
}

public class Side {

    private ValueWithUnits length;

    public ValueWithUnits getLength() {
        return length;
    }

    public void setLength(ValueWithUnits length) {
        this.length = length;
    }
}

public class ValueWithUnits {

    private String unit;

    public String getUnit() {
        return unit;
    }

    public void setUnit(String unit) {
        this.unit = unit;
    }

    private BigDecimal magnitude;

    public BigDecimal getMagnitude() {
        return magnitude;
    }

    public void setMagnitude(BigDecimal magnitude) {
        this.magnitude = magnitude;
    }
}
